esphome/components/samsung/samsung.h

```diff
@@ -0,0 +1,204 @@
+#pragma once
+
+#include "esphome/components/climate/climate_mode.h"
+#include "esphome/components/climate_ir/climate_ir.h"
+
+namespace esphome {
+namespace samsung {
+
+#define GETBITS8(data, offset, size) (((data) & (((uint8_t) UINT8_MAX >> (8 - (size))) << (offset))) >> (offset))
+
+static const char *const TAG = "samsung.climate";
+static const uint32_t SAMSUNG_IR_FREQUENCY = 38000;
+static const int SAMSUNG_AIRCON1_HDR_MARK = 3000;
+static const int SAMSUNG_AIRCON1_HDR_SPACE = 9000;
+static const int SAMSUNG_AIRCON1_BIT_MARK = 500;
+static const int SAMSUNG_AIRCON1_ONE_SPACE = 1500;
+static const int SAMSUNG_AIRCON1_ZERO_SPACE = 500;
+static const int SAMSUNG_AIRCON1_MSG_SPACE = 2000;
```

Comment:
```suggestion
// All timing constants are in microseconds (us)
static const uint16_t SAMSUNG_AIRCON1_HDR_MARK_US = 3000;
static const uint16_t SAMSUNG_AIRCON1_HDR_SPACE_US = 9000;
static const uint16_t SAMSUNG_AIRCON1_BIT_MARK_US = 500;
static const uint16_t SAMSUNG_AIRCON1_ONE_SPACE_US = 1500;
static const uint16_t SAMSUNG_AIRCON1_ZERO_SPACE_US = 500;
static const uint16_t SAMSUNG_AIRCON1_MSG_SPACE_US = 2000;
```

I think they're in microseconds and its fine to keep them as uint16_t. Is there specific reason not to do it?

https://github.com/esphome/esphome/blob/387bde665e3ae5d4b8d5a68299abd6032c4291b8/esphome/components/remote_base/remote_base.h#L27-L32